### PR TITLE
Tools: autotest: Added test for checking correct local target for Copter

### PR DESCRIPTION
**Pull request by rajat2004**
Recently, in PR #10753, support for using SET_MESSAGE_INTERVAL to request POSITION_TARGET_LOCAL_NED was added for Copter. This is a continuation of that to add tests for checking whether the vehicle is receiving the correct local target or not.
More tests such as to check the velocity, both position & velocity, and also to ensure the message is not sent outside of Guided mode will be added.

Suggestions for naming conventions are very much required!!

### CODE REVIEW

File: Tools/autotest/arducopter.py [hunk 1]
```diff
@@ -2170,6 +2170,41 @@ def fly_guided_move_local(self, x, y, z_up, timeout=100):
...
+                0, # afx
+                0, # afy
+                0, # afz
+                0, # yaw
+                0, # yawrate
+            )
+        m = self.mav.recv_match(type='POSITION_TARGET_LOCAL_NED', blocking=True)
+        self.progress("Received local target: %s" % str(m))
+        
+        if m.type_mask == 0xFFF8 or m.type_mask == 0x0FF8:
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Invert this check, move the corresponding exception up to here, unindent what was the body of the if clause...

File: Tools/autotest/arducopter.py [hunk 2]
```diff
@@ -2170,6 +2170,41 @@ def fly_guided_move_local(self, x, y, z_up, timeout=100):
...
+            if not x == m.x and y == m.y and z_up == -(m.z):
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Please add parenthesis to make clear the binding tightness.

File: Tools/autotest/arducopter.py [hunk 3]
```diff
@@ -2170,6 +2170,41 @@ def fly_guided_move_local(self, x, y, z_up, timeout=100):
         if z_achieved - z_up > 1:
             raise NotAchievedException("Did not achieve z position: want=%f got=%f" % (z_up, z_achieved))
 
+    def test_guided_local_target(self, x, y, z_up):
+        """ Check target position being received by vehicle """
+        # set POSITION_TARGET_LOCAL_NED message rate using SET_MESSAGE_INTERVAL
+        self.progress("Setting local target in NED: (%f, %f, %f)" % (x, y, -z_up))
+        self.progress("Setting rate to 1 Hz")
+        self.set_message_rate_hz(85, 1)
```
Comment (peterbarker) [inline Tools/autotest/arducopter.py]: Should be using the constant from `mavutil.mavlink` here

**Comment by rajat2004** [comment]
@peterbarker Changes done. Should I add the following tests too in the same function?

> More tests such as to check the velocity, both position & velocity, and also to ensure the message is not sent outside of Guided mode will be added.

**Comment by rajat2004** [comment]
What should be the error tolerance? Currently, the build is failing due to a difference of 0.01m in the z component

**Comment by rajat2004** [comment]
I have kept the error as 0.1m currently

**Comment by peterbarker** [comment]
On Sun, 31 Mar 2019, Rajat Singhal wrote:

> What should be the error tolerance? Currently, the build is failing due to a difference of 0.01m in the z component

We only work to the cm in a lot of the position control code, so this 
isn't unexpected.

**Comment by peterbarker** [comment]
On Sun, 31 Mar 2019, Rajat Singhal wrote:

> @peterbarker Changes done. Should I add the following tests too in the same function?
>
>       More tests such as to check the velocity, both position & velocity, and also to ensure the message is not sent outside of Guided mode will be
>       added.

Sounds good!

Did you want to get this merged as-is first?  Well... once it passes, 
anyway ;-)

**Comment by rajat2004** [comment]
@peterbarker Not as it, just to make sure of the logic, then I can easily add the other tests with the proper structure

**Comment by rajat2004** [comment]
Sorry for not updating for so many days
@peterbarker I have added the tests but I'm not sure whether this is the correct method for testing the velocity targets.

**Comment by peterbarker** [comment]
Please rebase.

**Comment by rajat2004** [comment]
@peterbarker Done

**Comment by rajat2004** [comment]
The Semaphore build is failing due to it not being able to download some packages, not related to changes due to the commit. As for the other ones, they haven't even started for some reason.

@peterbarker Should I do some minor change and push again? Also, is the structure and method of testing proper or is there a better way to go about this?

**Comment by peterbarker** [comment]
@rajat2004 I've pushed this in by hand because our CI systems are playing up to a ridiculous level today.

Thanks for this contribution!

**Pull request closed without merging**